frontend/src/lib/StaticGraphParser.ts

```diff
@@ -290,9 +290,16 @@ export function createGraph(workflow: Workflow): dagre.graphlib.Graph {
  */
 export function transitiveReduction(graph: dagre.graphlib.Graph): dagre.graphlib.Graph | undefined {
   // safeguard against too big graphs
-  if (!graph || graph.edgeCount() > 1000 || graph.nodeCount() > 1000) {
+  if (!graph) {
+    console.log("Error: Graph not provided.");
```

Comment:
I think throwing an exception and showing an error dialog on the UI makes this error more apparent. Not everybody goes to developer console on browser if they encounter such issue. How do you think about using this function? https://github.com/kubeflow/pipelines/blob/270257512a0f217b598bf82ce6c5a283dcc458f4/frontend/src/pages/Page.tsx#L83

Reply:
Sounds good. My only question is as follows: given that showErrorDialog is a method of Page, and we are not subclassing Page in this file, do you have a recommended approach for accessing showErrorDialog? All other invocations of showErrorDialog take place in classes that extend Page. Would it be strange to import and instantiate Page for the sole purpose of invoking showErrorDialog?

Reply:
Since `RunDetail` and `PipelineDetail` pages are the two callers of this function: https://github.com/search?q=repo%3Akubeflow%2Fpipelines%20transitiveReduction&type=code, you can throw exception in `transitiveReduction()`, then catch them in `RunDetail` and `PipelineDetail` pages. Once you are in `RunDetail` and `PipelineDetail` pages, you can call `showErrorDialog`.

Reply:
Thanks for the guidelines, @zijianjoy! Really appreciate it.

Pushed up the changes. Happy to make any other modifications you think are necessary.

We ran into the following failures when executing the unit tests locally. It doesn't look related to me, but I'm not sure.
```
Summary of all failing tests
 FAIL  src/pages/RunDetails.test.tsx (33.131 s)
  ● RunDetails › calls retry API when retry dialog is confirmed

    TypeError: Cannot read property 'onClick' of undefined

      297 |     const call = updateDialogSpy.mock.calls[0][0];
      298 |     const confirmBtn = call.buttons.find((b: any) => b.text === 'Retry');
    > 299 |     await confirmBtn.onClick();
          |                      ^
      300 |     expect(retryRunSpy).toHaveBeenCalledTimes(1);
      301 |     expect(retryRunSpy).toHaveBeenLastCalledWith(testRun.run!.id);
      302 |   });

      at Object.onClick (src/pages/RunDetails.test.tsx:299:22)

  ● RunDetails › shows an error dialog when retry API fails

    TypeError: Cannot read property 'onClick' of undefined

      325 |     const call = updateDialogSpy.mock.calls[0][0];
      326 |     const confirmBtn = call.buttons.find((b: any) => b.text === 'Retry');
    > 327 |     await confirmBtn.onClick();
          |                      ^
      328 |     expect(updateDialogSpy).toHaveBeenLastCalledWith(
      329 |       expect.objectContaining({
      330 |         content: 'Failed to retry run: test-run-id with error: ""mocked error""',

      at Object.onClick (src/pages/RunDetails.test.tsx:327:22)

  ● RunDetails › calls terminate API when terminate dialog is confirmed

    TypeError: Cannot read property 'onClick' of undefined

      377 |     const call = updateDialogSpy.mock.calls[0][0];
      378 |     const confirmBtn = call.buttons.find((b: any) => b.text === 'Terminate');
    > 379 |     await confirmBtn.onClick();
```

Reply:
It is likely due to the mock test run is empty in this line: https://github.com/kubeflow/pipelines/blob/a2d1f89aa1274b6b57770151043bbf6794f74a84/frontend/src/pages/RunDetails.test.tsx#L143. You will need to fulfill it with some valid mock data, otherwise the UI will show the error dialog, which is expected because that is what your PR is trying to achieve.

Reply:
Copy that. Thanks, @zijianjoy! I've been slammed with other stuff but will try to resolve tomorrow. Thanks for your patience!